Firestore/Source/Public/FIRDocumentSnapshot.h

```diff
@@ -180,10 +180,22 @@ NS_SWIFT_NAME(QueryDocumentSnapshot)
 /**
  * Retrieves all fields in the document as an `NSDictionary`.
  *
+ * Server-provided timestamps that have not yet been set to their final value will be returned as
+ * `NSNull`. You can use `dataWithOptions()` to configure this behavior.
+ *
  * @return An `NSDictionary` containing all fields in the document.
  */
 - (NSDictionary<NSString *, id> *)data;
 
+/**
+ * Retrieves all fields in the document as a `Dictionary`.
+ *
+ * @param options `SnapshotOptions` to configure how data is returned from the snapshot (e.g. the
+ *      desired behavior for server timestamps that have not yet been set to their final value).
+ * @return A `Dictionary` containing all fields in the document.
+ */
+- (NSDictionary<NSString *, id> *)dataWithOptions:(FIRSnapshotOptions *)options;
```

Comment:
Please add a swift-language usage of this API to verify that it looks like what we want after conversion to swift.

Reply:
Done. Also added an NSSwiftName for FIRServerTimestampBehavior to make the type cleaner when it shows up in code completion.

Reply:
data(options:) likely. @ryanwilson, thoughts?

Reply:
Makes sense. The fluency argument is a good one. That makes this change good as-is.